cmd/ddev/cmd/addon-search.go

```diff
@@ -0,0 +1,106 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/ddev/ddev/pkg/ddevapp"
+	"github.com/ddev/ddev/pkg/github"
+	"github.com/ddev/ddev/pkg/globalconfig"
+	"github.com/ddev/ddev/pkg/output"
+	"github.com/ddev/ddev/pkg/styles"
+	"github.com/ddev/ddev/pkg/util"
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/jedib0t/go-pretty/v6/text"
+	"github.com/spf13/cobra"
+)
+
+// AddonSearchCmd is the "ddev add-on search" command
+var AddonSearchCmd = &cobra.Command{
+	Use:   "search <search-term>",
+	Args:  cobra.ExactArgs(1),
+	Short: "Search available DDEV add-ons",
+	Long:  `Search available DDEV add-ons by name or description.`,
+	Example: `ddev add-on search redis
+ddev add-on search database
+ddev add-on search "solr search"`,
```

Comment:
Can we allow sending multiple args instead? e.g. `ddev add-on search solr search` instead of having to put quotes around it?

This would also allow for potential future enhancements where each argument is a separate search term, e.g. `ddev add-on search solr search` could find any instance of "solr" and any instance of "search", while `ddev add-on search "solr search"` might explicitly only find instances of the full string "solr search" and not "solr" or "search" individually.